torch/csrc/onnx/diagnostics/generated/rules.h

```diff
@@ -0,0 +1,48 @@
+#pragma once
+
+/**
+ * GENERATED CODE - DO NOT EDIT DIRECTLY
+ * This file is generated by gen_diagnostics.py.
+ * See tools/onnx/gen_diagnostics.py for more information.
+ *
+ * Diagnostic rules for PyTorch ONNX export.
+ */
+
+namespace torch {
+namespace onnx {
+namespace diagnostics {
+
+enum class Rule : uint32_t {
+  /**
+   * @brief Node is missing ONNX shape inference.
+   */
+  kNodeMissingOnnxShapeInference,
+
+  /**
+   * @brief Missing symbolic function for custom PyTorch operator, cannot
+   * translate node to ONNX.
+   */
+  kMissingCustomSymbolicFunction,
+
+  /**
+   * @brief Missing symbolic function for standard PyTorch operator, cannot
+   * translate node to ONNX.
+   */
+  kMissingStandardSymbolicFunction,
+
+  /**
+   * @brief Operator is supported in newer opset version.
+   */
+  kOperatorSupportedInNewerOpsetVersion,
+};
+
+static constexpr const char* const kPyRuleNames[] = {
+    "node_missing_onnx_shape_inference",
+    "missing_custom_symbolic_function",
+    "missing_standard_symbolic_function",
+    "operator_supported_in_newer_opset_version",
```

Comment:
Do we need to repeat the names of the rules here?

Reply:
Haven't found a good way for c++ to turn enum to string other than this, so ...